ldap/servers/slapd/ch_malloc.c

```diff
@@ -349,3 +349,57 @@ slapi_ct_memcmp(const void *p1, const void *p2, size_t n)
     }
     return result;
 }
+
+#define SYS_CACHELINE_SIZE "/sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size"
+static
+size_t cache_line_size() 
+{
+    PRFileDesc *prfd;
+    char buf[10] = {0};
+    size_t i = 0;
+
+    prfd = PR_Open(SYS_CACHELINE_SIZE, PR_RDONLY, SLAPD_DEFAULT_FILE_MODE);
+    if (PR_Read(prfd, buf, sizeof (buf)) < 0) {
+        slapi_log_err(SLAPI_LOG_ERR, "cache_line_size", "Can not read %s\n", SYS_CACHELINE_SIZE);
+        PR_Close(prfd);
+        goto done;
+    }
+    PR_Close(prfd);
+    i = atoi(buf);
+
+done:
+    if ((i == 0) || (i > 64)) {
+        slapi_log_err(SLAPI_LOG_INFO, "cache_line_size", "unexpected cpu cache line size => assign 64\n");
+        i = 64;
+    }
+    return i;
+}
+
+pthread_mutex_t*
+slapi_pthread_mutex_alloc(int type)
+{
+    int32_t cacheLineSiz = cache_line_size();
```

Comment:
We shouldn't call this every alloc because the value is effectively static. Is there a way we could make this a global and just call detect_cache_line_size once in main()?

Reply:
+100